iandavis/answers.js

```diff
@@ -0,0 +1,206 @@
+/* SECTION 1*/
+
+
+//a
+"1" == 1;
+//true converts string "1" to number 1;
+
+//b 
+"1" === 1;
+//false no conversion, no equality
+
+//c
+x == 'x'
+// only true if x is a number unless Infinity, -Infinity, or NaN; false if boolean
+
+//d 
+x == (x + '')
+//true unless x = NaN, undefined, Boolean
+
+//e
+'' == ' '
+//true
+
+//f
+x = true;
+//assignment of x to true
+
+//g
+var x; x == undefined
+//true, a variable without a defined value given the value undefined
+
+//h
+'9'<'10'
+//intuitively true, actually false because JS goes by the first character of the string.  So, '090000000000'<'1000' is true, but '9'<'10'
+
+//i
+typeof x + 1 === "number"
+//Always false because "typeof x" has precedence.  typeof x evaluates to a string which is then concatenated with 1 resulting in "string1", "number1", "undefined1", or "boolean1"
+
+//j
+typeof x % 2 === "number"
+//Always false for basically the same reason as above (i), except the result is always NaN because % is operating of the string that results from "typeof x"
+
+//k
+typeof (x % 2) ==== "number"
+// always true because any x value % 2 will result in a "number" type
+
+//l
+x++ == ++x
+//true if x is Infinity or -Infinity because any operation on Infinity results in Infinity again. false if x is a string (always results in NaN). false for any number because the postfix increment operator increments after returning a value, while the prefix does so before returning a value.  Used in this way the prefix and postfix operators increment _away_ from each other, causing the value to never equal itself in this instance.
+
+//m
+++x == x++
+//true if x equals any number, including Infinity, though not NaN, because x is incremented left of the equality operator, then remains the same right of the equality operator. Also true for strings that are quoted numbers, which are converted to numbers. Also true if x is a boolean value.
+
```

Comment:
Ian, you're right where I was wrong!  I overlooked Infinity in my solution, and that is indeed a case where Infinity++ == ++Infinity, essentially because Infinity == Infinity+2.  Good job!